src/scripts/scaling/scalingAction.sh

```diff
@@ -16,10 +14,328 @@ scaling_size=1
+      in_cluster_startup="True"
+    fi
+  else
+cat > cmds-$$.py << INPUT
+import sys, json
+outer_loop_must_break = False
+for i in json.load(sys.stdin)["items"]:
+  j = i["spec"]["clusters"]
+  for index, cs in enumerate(j):
+    if j[index]["clusterName"] == "$wls_cluster_name":
+      outer_loop_must_break = True
+      print True
+      break
+if outer_loop_must_break == False:
+  print False
+INPUT
+in_cluster_startup=`echo ${DOMAIN} | python cmds-$$.py`
+  fi
+  echo "$in_cluster_startup"
+}
+
+# Gets the current replica count of the cluster
+function get_num_ms_in_cluster() {
+  local num_ms
+  if [ -x "$(command -v jq)" ]; then
+  local numManagedServersCmd="(.items[].spec.clusters[] | select (.clusterName == \"${wls_cluster_name}\") | .replicas)"
+  num_ms=$(echo "${DOMAIN}" | jq "${numManagedServersCmd}")
+  else
+cat > cmds-$$.py << INPUT
+import sys, json
+for i in json.load(sys.stdin)["items"]:
+  j = i["spec"]["clusters"]
+  for index, cs in enumerate(j):
+    if j[index]["clusterName"] == "$wls_cluster_name":
+      print j[index]["replicas"]
+INPUT
+  num_ms=`echo ${DOMAIN} | python cmds-$$.py`
+  fi
+
+  if [ "${num_ms}" == "null" ]; then
+    num_ms=0
+  fi
+
+  echo "$num_ms"
+}
+
+# Gets the replica count at the Domain level
+function get_num_ms_domain_scope() {
+  local num_ms
+  if [ -x "$(command -v jq)" ]; then
+    num_ms=$(echo "${DOMAIN}" | jq -r '.items[].spec.replicas' )
+  else
+cat > cmds-$$.py << INPUT
+import sys, json
+for i in json.load(sys.stdin)["items"]:
+  print i["spec"]["replicas"]
+INPUT
+  num_ms=`echo ${DOMAIN} | python cmds-$$.py`
+  fi
+
+  if [ "${num_ms}" == "null" ]; then
+    num_ms=0
+  fi
+
+  echo "$num_ms"
+}
+
+#
+# Function to get minimum replica count for cluster
+# $1 - Domain resource in json format
+# $2 - Name of the cluster
+# $3 - Return value containing minimum replica count
+#
+function get_min_replicas {
+  local domainJson=$1
+  local clusterName=$2
+  local __result=$3
+
+  eval $__result=0
+  minReplicaCmd="(.status.clusters[] | select (.clusterName == \"${clusterName}\")) \
+    | .minimumReplicas"
+  minReplicas=$(echo ${domainJson} | jq "${minReplicaCmd}")
+  eval $__result=${minReplicas}
+}
+
+# Get the current replica count for the WLS cluster if defined in the CRD's Cluster
+# configuration.  If WLS cluster is not defined in the CRD then return the Domain
+# scoped replica value, if present.  Returns replica count = 0 if no replica count found.
+# args:
+# $1 "True" if WLS cluster configuration defined in CRD, "False" otherwise
+# $2 Custom Resource Domain
+function get_replica_count() {
+  local in_cluster_startup="$1"
+  local DOMAIN="$2"
+  local num_ms
+  if [ "$in_cluster_startup" == "True" ]
+  then
+    trace "$wls_cluster_name defined in clusters"
+    num_ms=$(get_num_ms_in_cluster "$DOMAIN")
+  else
+    trace "$wls_cluster_name NOT defined in clusters"
+    num_ms=$(get_num_ms_domain_scope "$DOMAIN")
+  fi
+  echo "$num_ms"
```

Comment:
This should return min_replica_count if discovered value is less than the min

Reply:
Added code to return 'min_replica_count' if min_replica_ count is higher than the discovered value.